src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/RuleRunner.cs

```diff
@@ -63,7 +65,11 @@ private void InitializeRules()
             foreach (Type type in GetType().Assembly.GetTypes())
```

Comment:
Unrelated to this PR, but this feels like unnecessary runtime expense in every execution of this API.  You can know statically this answer.  You're making users pay (many times over) to save yourself from making a static list of types.  If you wanted to avoid that static list this could be a source generator.

Reply:
IMHO we shouldn't pay for extensibility until we need it.  It's fine to have an idea of how you'd do that when it comes but until then I'd opt for keep-it-simple.